app/assets/javascripts/components/plots/forms/plot_form.jsx

```diff
@@ -0,0 +1,348 @@
+        <ul>
+          {data.map(({ name, id }) => (
+            <li key={id}>
+              {name + ' '}
+              <i className='fa fa-times' aria-hidden='true'
+                 onClick={() => this.removePlotRefData(id)}>
+              </i>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+
+    render () {
+      const { toggleEditing, selectedManifest, handleSave, isNewPlot, plot, consignment } = this.props;
+
+      return (
+        <div className='plot-form-container'>
+          <ButtonBar
+            className='actions'
+            buttons={[
+              {
+                click: handleSave,
+                icon: 'floppy-o',
+                label: 'save'
+              }, {
+                click: () => toggleEditing(false),
+                icon: 'ban',
+                label: 'cancel'
+              }
+            ]}
+          />
+          {isNewPlot ? (
+            <div>
+              {this.plotTypeSelector()}
+              {this.manifestSelector()}
+            </div>
+          ) : (
+            <div>
+              <div>{`Plot Type: ${plotTypeLabel}`}</div>
+              <div>{`Manifest: ${selectedManifest.name}`}</div>
+            </div>
+          )}
+          <fieldset>
+            <legend>{plotTypeLabel}</legend>
+            {this.formFields()}
+          </fieldset>
+          <fieldset style={{ marginBottom: 10 }}>
+            <legend>Data</legend>
+            {this.dataRefList()}
+            {this.dataFormFields()}
+            <input type='button' value='Add' onClick={() => this.addDataRef()} />
+          </fieldset>
+          <Plot plot={plot} consignment={consignment}/>
+        </div>
+      );
+    }
+  };
+};
+
+/*
+input components subscribe to a property from the plot configuration state.
+eg. subscribePlotInputField(['layout', 'height'])(IntegerField('Height: '))
+the nested value for { layout: { height: value } } is injected into the input component
+*/
+export const subscribePlotInputField = (plotProperty = []) => (WrappedInput) => (props) => {
+  let { value, plot, consignment, updatePlot, ...passThroughProps } = props;
+
+  let currentValue = plotProperty.reduce((currVal, currProperty) => {
+    try {
+      return currVal[currProperty];
+    } catch (e) {
+      return null;
+    }
+  }, plot);
+
+  if (typeof currentValue !== 'string' && isNaN(currentValue)) {
+    currentValue = '';
+  }
+
+  const update = (value) => {
+    let updatedPlot = { ...plot };
+
+    let ref = updatedPlot;
+    const finalPropertyIndex = plotProperty.length - 1;
+    for (let i = 0; i < finalPropertyIndex; i++) {
+      ref = ref[plotProperty[i]];
+    }
+
+    ref[plotProperty[finalPropertyIndex]] = value;
+
+    updatePlot(updatedPlot);
+  };
+
+  return (
+    <WrappedInput
+      {...passThroughProps}
+      consignment={consignment}
+      value={currentValue}
+      onChange={update}
+    />
+  );
+};
+
+const consignmentKeysByType = (type) => (consignment) => {
```

Comment:
Since there is a Consignment model, I'd consider moving this stuff to a method on that consignment class, since it is probably generally useful outside of this context.